samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -39,60 +37,58 @@
-   * @param streamApplication the user-implemented {@link StreamApplication} that creates the logical DAG
-   * @param runner the {@link ApplicationRunner} to get the mapping between logical and physical streams
+   * Constructs an adaptor task to run the user-implemented {@link org.apache.samza.operators.StreamGraphImpl}.
+   * @param serializedGraph the serialized version of user-implemented {@link org.apache.samza.operators.StreamGraphImpl} that includes the logical DAG
    * @param clock the {@link Clock} to use for time-keeping
    */
-  public StreamOperatorTask(StreamApplication streamApplication, ApplicationRunner runner, Clock clock) {
-    this.streamApplication = streamApplication;
-    this.runner = runner;
+  public StreamOperatorTask(SerializedStreamGraph serializedGraph, Clock clock) {
+    this.serializedGraph = serializedGraph;
     this.clock = clock;
   }
 
-  public StreamOperatorTask(StreamApplication application, ApplicationRunner runner) {
-    this(application, runner, SystemClock.instance());
+  public StreamOperatorTask(SerializedStreamGraph streamGraph) {
+    this(streamGraph, SystemClock.instance());
   }
 
   /**
    * Initializes this task during startup.
    * <p>
-   * Implementation: Initializes the user-implemented {@link StreamApplication}. The {@link StreamApplication} sets
-   * the input and output streams and the task-wide context manager using the {@link StreamGraphImpl} APIs,
+   * Implementation: Initializes the user-implemented {@link org.apache.samza.operators.StreamGraphImpl}.
+   * The {@link org.apache.samza.operators.StreamGraphImpl} sets the input and output streams and the task-wide
+   * context manager using the {@link org.apache.samza.operators.StreamGraphImpl} APIs,
    * and the logical transforms using the {@link org.apache.samza.operators.MessageStream} APIs. It then uses
-   * the {@link StreamGraphImpl} to create the {@link OperatorImplGraph} corresponding to the logical DAG.
+   * the serialized instance of {@link org.apache.samza.operators.StreamGraphImpl} to create the {@link OperatorImplGraph}
+   * corresponding to the logical DAG.
    *
    * @param config allows accessing of fields in the configuration files that this StreamTask is specified in
    * @param context allows initializing and accessing contextual data of this StreamTask
    * @throws Exception in case of initialization errors
    */
   @Override
   public final void init(Config config, TaskContext context) throws Exception {
-    StreamGraphImpl streamGraph = new StreamGraphImpl(runner, config);
-    // initialize the user-implemented stream application.
-    this.streamApplication.init(streamGraph, config);
 
     // get the user-implemented context manager and initialize it
-    this.contextManager = streamGraph.getContextManager();
-    if (this.contextManager != null) {
-      this.contextManager.init(config, context);
+    if (serializedGraph.getContextManager() != null) {
```

Comment:
IIUC this is the same as before?

Reply:
Yep, shuffled during multiple rounds of merge. Revert to original.